pkg/offering/k8s/organization/organization.go

```
@@ -81,3 +139,116 @@ func (organization *Organization) ReconcileManagers(instance *current.Organizati
+		return req, &profile, nil
+	}
+	log.Info("after ten attempts, still can't get the information of ca running properly.\n")
+	return nil, nil, fmt.Errorf("exceed the maximum number of retry %d", maxRetryCount)
+}
+
+func (organization *Organization) DoEnroll(ctx context.Context, instance *current.Organization) {
+	log.Info("starting do enroll for admin and org")
+	enrollment, profileStr, err := organization.enrollArg(ctx, instance.GetName(), instance.GetName(), instance.Spec.Admin, instance.Spec.Token)
+	if err != nil {
+		return
+	}
+	certBytes, _ := enrollment.GetCATLSBytes()
+	caClient := enroller.NewFabCAClient(enrollment, adminCertPath, nil, certBytes)
+	certEnroller := enroller.New(enroller.NewSWEnroller(caClient))
+
+	resp, err := config1.GenerateCrypto(certEnroller)
+	if err != nil {
+		log.Error(err, "GenerateCrypto error")
+		return
+	}
+
+	s := v1.Secret{}
+	s.Name = instance.GetName() + "-msg-crypto"
+	s.Namespace = instance.GetName()
+	s.Data = make(map[string][]byte)
+	s.Data["admin-signcert"] = resp.SignCert
+	s.Data["admin-keystore"] = resp.Keystore
+
+	s.Data["org-ca-signcert"] = []byte(profileStr.CA.SignCerts)
+	s.Data["org-tlsca-signcert"] = []byte(profileStr.TLSCA.SignCerts)
+	s.Data["msg-config"] = []byte(mspConfig)
+
+	if err = organization.BaseOrganization.Client.Create(ctx, &s); err != nil {
+		if !errors2.IsAlreadyExists(err) {
+			log.Error(err, "create secret %s error.", s.GetName())
+			return
+		}
+		log.Info("secret %s already exists, try to update...\n")
+
+		if organization.BaseOrganization.Client.Get(ctx, types.NamespacedName{Namespace: instance.GetName(), Name: s.GetName()}, &s); err != nil {
+			log.Error(err, "get secret %s error", s.GetName())
+			return
+		}
+
+		s.Data["signcert"] = resp.SignCert
+		s.Data["keystore"] = resp.Keystore
+		s.Data["org-ca-signcert"] = []byte(profileStr.CA.SignCerts)
+		s.Data["org-tlsca-signcert"] = []byte(profileStr.TLSCA.SignCerts)
+		s.Data["msg-config"] = []byte(mspConfig)
+
+		// ignore error
```

IMO,use `_ = xxx()` to ignore error